prusti-viper/src/encoder/mir_encoder/place_encoding.rs

```diff
@@ -53,6 +53,26 @@ impl<'tcx> PlaceEncoding<'tcx> {
         }
     }
 
+    /// Ok(e) => the same result try_into_expr would give
+    /// Err(e) => the array base expression
```

Comment:
Please don't use the error variant for things that are not actually errors. Define a dedicated enum, or see if you can make the uses of this function just do `try_into_array_base`, returning an `Option`, otherwise falling back to `try_into_expr`.

Reply:
tried to keep the duplication with `try_into_expr` in bounds, added a separate `ExprOrArrayBase` enum.